suite-native/module-add-coin/src/hooks/useAddCoinAccount.ts

```diff
@@ -0,0 +1,235 @@
+import { useCallback, useState } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+
+import { useNavigation } from '@react-navigation/native';
+import { A, pipe } from '@mobily/ts-belt';
+
+import { AccountType, NetworkSymbol, Network } from '@suite-common/wallet-config';
+import {
+    AccountsRootState,
+    DeviceRootState,
+    LIMIT,
+    selectDevice,
+    selectDeviceAccounts,
+} from '@suite-common/wallet-core';
+import { useAlert } from '@suite-native/alerts';
+import {
+    addNetworkAccountThunk,
+    selectAreTestnetsEnabled,
+    selectDiscoverySupportedNetworks,
+    NORMAL_ACCOUNT_TYPE,
+} from '@suite-native/discovery';
+import { useTranslate } from '@suite-native/intl';
+import { useOpenLink } from '@suite-native/link';
+import {
+    RootStackParamList,
+    RootStackRoutes,
+    StackNavigationProps,
+} from '@suite-native/navigation';
+
+const availableNetworkAccountTypes: Record<NetworkSymbol, AccountType[]> = {
+    btc: [NORMAL_ACCOUNT_TYPE, 'taproot', 'segwit', 'legacy'],
+    ltc: [NORMAL_ACCOUNT_TYPE, 'segwit', 'legacy'],
+    btg: [NORMAL_ACCOUNT_TYPE, 'segwit', 'legacy'],
+    dgb: [NORMAL_ACCOUNT_TYPE, 'segwit', 'legacy'],
+    vtc: [NORMAL_ACCOUNT_TYPE, 'segwit', 'legacy'],
+    ada: [NORMAL_ACCOUNT_TYPE],
+    sol: [NORMAL_ACCOUNT_TYPE],
+    eth: [NORMAL_ACCOUNT_TYPE],
+    etc: [NORMAL_ACCOUNT_TYPE],
+    xrp: [NORMAL_ACCOUNT_TYPE],
+    bch: [NORMAL_ACCOUNT_TYPE],
+    dash: [NORMAL_ACCOUNT_TYPE],
+    doge: [NORMAL_ACCOUNT_TYPE],
+    nmc: [NORMAL_ACCOUNT_TYPE],
+    zec: [NORMAL_ACCOUNT_TYPE],
+    test: [NORMAL_ACCOUNT_TYPE],
+    regtest: [NORMAL_ACCOUNT_TYPE],
+    tsep: [NORMAL_ACCOUNT_TYPE],
+    tgor: [NORMAL_ACCOUNT_TYPE],
+    thol: [NORMAL_ACCOUNT_TYPE],
+    txrp: [NORMAL_ACCOUNT_TYPE],
+    tada: [NORMAL_ACCOUNT_TYPE],
+    dsol: [NORMAL_ACCOUNT_TYPE],
+};
```

Comment:
`as const satisfies Record<NetworkSymbol, AccountType[]>` would be better IMO. like that you cannot overwrite neither the keys or values. So the object is freezed in this shape and TypeScript knows exactly which coin has which account types.

Reply:
Also `AccountType[]` could be changed to  `[AccountType, ...AccountType[]]` So there is forced that the account type array cannot be empty.

Reply:
updated